### DISCUSSION TITLE
Package not building when installed on vercel deployment

### DISCUSSION THREAD

**Discussion by Dakauann** [vercel/community#1603] [Help]
### Summary

Im trying to host a website to documment my library thats published on npm, bdw when the deployment starts it shows that it just built all the npm packages and installed them, logs below:

Running "install" command: `npm i`...
20:17:25.113 |  
20:17:25.114 | added 172 packages, and audited 173 packages in 24s
20:17:25.114 |  
20:17:25.114 | 31 packages are looking for funding
20:17:25.114 | run `npm fund` for details
20:17:25.116 |  
20:17:25.116 | found 0 vulnerabilities
20:17:25.139 | Detected Next.js version: 13.1.2
20:17:25.143 | Detected `package-lock.json` generated by npm 7+...
20:17:25.143 | Running "npm run build"
20:17:25.564 |  
20:17:25.564 | > lumin-ui@0.1.0 build
20:17:25.564 | > next build
20:17:25.564 |  
20:17:26.134 | Attention: Next.js now collects completely anonymous telemetry regarding usage.
20:17:26.135 | This information is used to shape Next.js' roadmap and prioritize features.
20:17:26.135 | You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
20:17:26.135 | https://nextjs.org/telemetry
20:17:26.135 |  
20:17:26.302 | info  - Linting and checking validity of types...
20:17:31.111 | info  - Creating an optimized production build...
20:17:39.874 | Failed to compile.

 but when it starts the actual code, it doesnt found the files that has just been built here:  lumin-ui@0.1.0 build

and then throws this error:
Error: Cannot find module '/vercel/path0/node_modules/lumin-ui/dist/cjs/index.js'. Please verify that the package.json has a valid "main" entry

### Additional information

those are my npm library package build scripts: 
 "build": "rm -rf dist/ && npm run build:esm && npm run build:cjs && echo 'Build finished!'",
  "build:esm": "tsc",
  "build:cjs": "tsc --module CommonJS --outDir dist/cjs",
  "install": "npm run build"

**Comment by amyegan** [comment]
I see the `Failed to compile` message in your build log, but the log doesn't give me enough info to pinpoint a cause. Are you able to share a minimal reproduction so I can take a closer look?

**Reply by Dakauann** [reply]
> I see the `Failed to compile` message in your build log, but the log doesn't give me enough info to pinpoint a cause. Are you able to share a minimal reproduction so I can take a closer look?

Hello, those are the steps to reproduce:
Create a typescript nextjs project
Install tailwind css
install my library (lumin-ui) via npm (https://github.com/Dakauann/lumin-ui)
Deploy it to vercel
